### PR TITLE
Fabs: new package

### PR DESCRIPTION
**Pull request by spacefrogg**
###### Description of changes

Adds a modern OpenAFS backup server option to the openafsServer module. The program is called FABS and makes use of existing tools. Thus, it depends on `kstart` for Kerberos ticket retrieval, which is also added as a new package.

Depends on #220176 for the exposure of the `afsdump_scan` tool that comes with openafs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/backup/fabs/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{ python3Packages
+, lib
+, fetchFromGitHub
+, perl
+, sqlite
+}:
+
+with python3Packages;
```
Comment (AndersonTorres) [inline pkgs/tools/backup/fabs/default.nix:8]:
```suggestion
```

Never use `with` this way.

File: pkgs/tools/backup/fabs/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  src = fetchFromGitHub {
+    owner = "openafs-contrib";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-ejAcCwrOWGX0zsMw224f9GTWlozNYC0gU6LdTk0XqH0=";
+  };
+
+  nativeBuildInputs = [
+    perl # pod2man
```
Comment (AndersonTorres) [inline pkgs/tools/backup/fabs/default.nix:23]:
```suggestion
    perl
```

File: pkgs/tools/security/kstart/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+, libkrb5
+, openafs
+, keyutils
+, enableSetPAG ? false
+}:
+
+let
+  pname = "kstart";
+in stdenv.mkDerivation (finalAttrs: {
+  inherit pname;
```
Comment (AndersonTorres) [inline pkgs/tools/security/kstart/default.nix:16]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "kstart";
```

File: pkgs/tools/security/kstart/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  version = "4.3";
+
+  src = fetchFromGitHub {
+    owner = "rra";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/tools/security/kstart/default.nix:21]:
```suggestion
    repo = "kstart";
```

File: pkgs/tools/security/kstart/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+, lib
+, autoreconfHook
+, pkgconfig
+, perl
+, fetchFromGitHub
+, libkrb5
+, openafs
+, keyutils
+, enableSetPAG ? false
+}:
```
Comment (AndersonTorres) [inline pkgs/tools/security/kstart/default.nix:11]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, autoreconfHook
, keyutils
, libkrb5
, openafs
, perl
, pkg-config
, enableSetPAG ? false
}:
```

File: pkgs/tools/security/kstart/default.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    rev = "release/${finalAttrs.version}";
+    hash = "sha256-MGWL4oNc0MZTGWqBEt2wRTkqoagiUTDrS0kz4ewbZZA=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig perl ];
+
+  buildInputs = [ libkrb5 keyutils openafs ];
+
+  configureFlags = (lib.optional enableSetPAG "--enable-setpag") ++
+                   [ "--enable-silent-rules" ];
```
Comment (AndersonTorres) [inline pkgs/tools/security/kstart/default.nix:31]:
```suggestion
  nativeBuildInputs = [ 
    autoreconfHook 
    perl
    pkg-config
  ];

  buildInputs = [ 
    keyutils 
    libkrb5 
    openafs 
  ];

  configureFlags =   [ 
    "--enable-silent-rules"
  ] 
  ++ (lib.optional enableSetPAG "--enable-setpag");
```

File: pkgs/tools/backup/fabs/default.nix [hunk 7]
```diff
@@ -0,0 +1,61 @@
+{ python3Packages
+, lib
+, fetchFromGitHub
+, perl
+, sqlite
+}:
```
Comment (AndersonTorres) [inline pkgs/tools/backup/fabs/default.nix:6]:
```suggestion
{ lib
, fetchFromGitHub
, perl
, python3
, sqlite
}:
```

File: pkgs/tools/backup/fabs/default.nix [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+with python3Packages;
+
+buildPythonApplication rec {
+  pname = "fabs";
+  version = "1.1";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "openafs-contrib";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/tools/backup/fabs/default.nix:17]:
```suggestion
    repo ="fabs";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Are the errors on Darwin relevant?

**Comment by spacefrogg** [comment]
They are not. It is not intended for use on Darwin. Can/Shall I restrict the platforms in a Python application?

**Comment by AndersonTorres** [comment]
Yes, you should - at least in this particular case.
Without it, Ofborg and Hydra will try to evaluate and fail, generating a spurious entry for the future Zero Hydra Failures event :)

I believe setting `meta.badPlatforms` is sufficient.

**Review by AndersonTorres — APPROVED** [review]

**Comment by spacefrogg** [comment]
I updated the package. Thanks for the guidance!

**Comment by spacefrogg** [comment]
This one is now ready to be merged.

File: pkgs/tools/backup/fabs/default.nix [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    hash = "sha256-ejAcCwrOWGX0zsMw224f9GTWlozNYC0gU6LdTk0XqH0=";
+  };
+
+  nativeBuildInputs = [
+    perl
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    alembic
+    dateutil
```
Comment (mweinelt) [inline pkgs/tools/backup/fabs/default.nix:26]: This is an alias to `python-dateutil` and using it breaks eval without aliases enabled.

Fix included in #225732.

Comment (spacefrogg) [reply pkgs/tools/backup/fabs/default.nix:26]: Thanks! How could I have detected that myself? Is the only way to check against the list of aliases manually? nixpkgs-review does not seem to disable aliases when it runs...

**Pull request merged**